cli/src/semgrep/config_resolver.py

```diff
@@ -226,6 +237,23 @@ def from_pattern_lang(
         valid, errors = cls._validate(config_dict)
         return cls(valid), errors
 
+    @classmethod
+    def from_rules_yaml(cls, config: str) -> Tuple["Config", Sequence[SemgrepError]]:
+        config_dict: Dict[str, YamlTree] = {}
+        errors: List[SemgrepError] = []
+
+        try:
+            resolved_config_key = "rules-yaml-string"
```

Comment:
what's this?

Reply:
A hack! It seems like the current config_resolver expects a dictionary so that each config can be accessed by a unique key. To fit that schema and avoid re-writing most of config_resolver, we just picked a key that would indicate what this config is (a stringified rules.yaml file).